src/Main.hh

```diff
@@ -73,12 +73,43 @@ public:
             });
         }
 
+        //exclude patterns, fill a array
+        split(this->exclude, ',', patterns);
+        
         // Transfer the list to dmenu
         for(auto &app : iteration_order) {
-            this->dmenu->write(app.second->name);
-            const std::string &generic_name = app.second->generic_name;
-            if(!exclude_generic && !generic_name.empty() && app.second->name != generic_name)
-                this->dmenu->write(generic_name);
+	    const std::string app_name = app.second->name;
+	    const std::string app_gen_name = app.second->generic_name;
+
+	    std::string lower_app_name = app_name;
+	    std::string lower_app_gen_name = app_gen_name;
+	    std::transform(lower_app_name.begin(), lower_app_name.end(), lower_app_name.begin(), ::tolower);
+	    std::transform(lower_app_gen_name.begin(), lower_app_gen_name.end(), lower_app_gen_name.begin(), ::tolower);
```

Comment:
Since this block does some work, put it (and the following loop) into an `if` that only runs if exclude patterns were specified?